src/platform/packages/shared/react/kibana_context/root/eui_provider.tsx

```diff
@@ -89,6 +98,16 @@ export const KibanaEuiProvider: FC<PropsWithChildren<KibanaEuiProviderProps>> =
   // colorMode provided by the `theme`.
   const colorMode = colorModeProp || themeColorMode;
 
+  const getUserProfile$ = userProfile?.getUserProfile$ ?? Rx.of;
```

Comment:
tiny suggestion to memo the observable, because now useObservable would unsubscribe and subscribe with every render

Reply:
@Dosant I take it you mean we should call a memoized version of `getUserProfile$`. Is this what you mean?
```diff
-  const getUserProfile$ = userProfile?.getUserProfile$ ?? Rx.of;
+  const getUserProfile$ = useMemo(
+    () => userProfile?.getUserProfile$ ?? Rx.of,
+    [userProfile?.getUserProfile$]
+  );
```